src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs

```diff
@@ -277,7 +277,7 @@ public ProjectContext Build()
 
                     if (platformDependency != null)
                     {
-                        platformLibrary = libraries[new LibraryKey(platformDependency.Value.Name)];
+                        libraries.TryGetValue(new LibraryKey(platformDependency.Value.Name), out platformLibrary);
```

Comment:
You are saying *everywhere* we create a projectcontext we should do validation? 

Why wouldn't we handle the validation in the common piece (project context creation), and use built in error mechanisms to communicate the validation failures?